src/purva/Shop.java

```diff
@@ -0,0 +1,101 @@
+package purva;
+
+public class Shop {
+	
+	int maggie;
+	int dosa;
+	int oilPackets;
+	int panipuri;
+	int masala;
+	
+	void setInitialQuantity(int maggieQ,int dosaQ,int oilPacketsQ,int panipuriQ,int masalaQ) {
+		maggie = maggieQ;
+		dosa = dosaQ;
+		oilPackets = oilPacketsQ;
+		panipuri = panipuriQ;
+		masala = masalaQ;
+		
+	}
+	
+	void outOfStock(int maggieQ,int dosaQ,int oilPacketsQ,int panipuriQ,int masalaQ) {
+		
+		/*maggie=maggieQ;
+		dosa=dosaQ;
+		oilPackets=oilPacketsQ;
+		panipuri=panipuriQ;
+		masala=masalaQ;*/
+		
+		if(maggieQ>=maggie) 
+			System.out.println("maggie is out of stock");
+		if(dosaQ >=dosa)
+			System.out.println("Dosa is out of stock");
+		if(oilPacketsQ >=oilPackets)
+		    System.out.println("oil packets are out of stock");
+		if(panipuriQ >=panipuri) 
+			System.out.println("panipuri is out of stock");
+		if(masalaQ >=masala)
+			System.out.println("masala is out of stock");
+		 
+	}
+	
+	void requestedQuantity(int maggieQ,int dosaQ,int oilPacketsQ,int panipuriQ,int masalaQ) {
+		
+		if(maggieQ <=maggie) {
+			maggie = maggie - maggieQ;
+		    System.out.println("customer purchased "+maggieQ+" and  Quantity of maggie in stock now is "+maggie);
+		}
+		else {
+			System.out.println("requested quantity pf maggie is "+maggieQ+" and we can provide is "+maggie+" packets");
+		    maggie=0;
```

Comment:
If request stock is not available mean we are setting direct it zero, 
what if second time your calling same method and let say you request 20 packs, then this looks not logically correct